Add cofidis3x, cofidis4x, ipwireinst & ipwire icons.

diasnath opened the pull request:
## Why are you adding this icons?
I'm adding/updating this icon(s) because ..

#### Help us identify yourself
- [ ] I'm working/collaborating with the brand directly and they have provided the icons.
- [ ] I'm associated with the brand and I've read all the brand icon’s guidelines.
- [x] I'm an individual and I've read all the brand icon’s guidelines.

Link to the brand guidelines: <Link>

## Checklist to add new icons
- [x] All icons have a corresponding entry in `db/payment_icons.yml`
- [x] I have followed the icon guidelines detailed in the [CONTRIBUTING.md](https://github.com/activemerchant/payment_icons/blob/master/CONTRIBUTING.md) file
- [x] I have optimized the icon with [SVGO](https://jakearchibald.github.io/svgomg/)
- [x] I am confident that all icons are clear and easy to read/understand
- [ ] I have provided a link to the brand icon’s brand guidelines whenever possible.
- [ ] I have attached a screenshot comparison with the example icon provided in guidelines
- [x] I recognize that if my icon is not approved by the Shopify Partners team it may not receive review nor merger.

If this pull request is not adding new icons, you can remove this checklist.

## Attach a screenshot of the icon along side the example Visa icon

![image](https://github.com/user-attachments/assets/a26d99d0-c85d-4314-9ca0-26792b65d27b)

#### Tips how to create a screenshot
We have found free online SVG editor https://www.freecodeformat.com/svg-editor.php very useful to create one. Here is a sample code for you to verify that you icon appears properly along side the placeholder.

```
<!-- Change background color if needed to showcase your icon better -->
<style> body { background: white; } </style>

<!-- DO NOT DELETE EXAMPLE -->
<svg viewBox="0 0 38 24" xmlns="http://www.w3.org/2000/svg" role="img" width="38" height="24" aria-labelledby="pi-visa"><title id="pi-visa">Visa</title><path opacity=".07" d="M35 0H3C1.3 0 0 1.3 0 3v18c0 1.7 1.4 3 3 3h32c1.7 0 3-1.3 3-3V3c0-1.7-1.4-3-3-3z"/><path fill="#fff" d="M35 1c1.1 0 2 .9 2 2v18c0 1.1-.9 2-2 2H3c-1.1 0-2-.9-2-2V3c0-1.1.9-2 2-2h32"/><path d="M28.3 10.1H28c-.4 1-.7 1.5-1 3h1.9c-.3-1.5-.3-2.2-.6-3zm2.9 5.9h-1.7c-.1 0-.1 0-.2-.1l-.2-.9-.1-.2h-2.4c-.1 0-.2 0-.2.2l-.3.9c0 .1-.1.1-.1.1h-2.1l.2-.5L27 8.7c0-.5.3-.7.8-.7h1.5c.1 0 .2 0 .2.2l1.4 6.5c.1.4.2.7.2 1.1.1.1.1.1.1.2zm-13.4-.3l.4-1.8c.1 0 .2.1.2.1.7.3 1.4.5 2.1.4.2 0 .5-.1.7-.2.5-.2.5-.7.1-1.1-.2-.2-.5-.3-.8-.5-.4-.2-.8-.4-1.1-.7-1.2-1-.8-2.4-.1-3.1.6-.4.9-.8 1.7-.8 1.2 0 2.5 0 3.1.2h.1c-.1.6-.2 1.1-.4 1.7-.5-.2-1-.4-1.5-.4-.3 0-.6 0-.9.1-.2 0-.3.1-.4.2-.2.2-.2.5 0 .7l.5.4c.4.2.8.4 1.1.6.5.3 1 .8 1.1 1.4.2.9-.1 1.7-.9 2.3-.5.4-.7.6-1.4.6-1.4 0-2.5.1-3.4-.2-.1.2-.1.2-.2.1zm-3.5.3c.1-.7.1-.7.2-1 .5-2.2 1-4.5 1.4-6.7.1-.2.1-.3.3-.3H18c-.2 1.2-.4 2.1-.7 3.2-.3 1.5-.6 3-1 4.5 0 .2-.1.2-.3.2M5 8.2c0-.1.2-.2.3-.2h3.4c.5 0 .9.3 1 .8l.9 4.4c0 .1 0 .1.1.2 0-.1.1-.1.1-.1l2.1-5.1c-.1-.1 0-.2.1-.2h2.1c0 .1 0 .1-.1.2l-3.1 7.3c-.1.2-.1.3-.2.4-.1.1-.3 0-.5 0H9.7c-.1 0-.2 0-.2-.2L7.9 9.5c-.2-.2-.5-.5-.9-.6-.6-.3-1.7-.5-1.9-.5L5 8.2z" fill="#142688"/></svg>

<!-- TODO: insert your icon here -->
<svg xmlns="http://www.w3.org/2000/svg" role="img" viewBox="0 0 38 24" width="38" height="24" aria-labelledby="pi-buckaroopaybybank"><title id="pi-buckaroopaybybank">Buckaroo Pay by Bank</title><path fill="#000" opacity=".07" d="M35 0H3C1.3 0 0 1.3 0 3v18c0 1.7 1.4 3 3 3h32c1.7 0 3-1.3 3-3V3c0-1.7-1.4-3-3-3z"/><path fill="#163255" fill-rule="evenodd" clip-rule="evenodd" d="M35 1c1.1 0 2 .9 2 2v18c0 1.1-.9 2-2 2H3c-1.1 0-2-.9-2-2V3c0-1.1.9-2 2-2h32"/><path fill="#CDF564" d="M7.602 6.324h1.316v3.844H7.602zm.613 1.727h1.363c.106 0 .2-.016.277-.04a.39.39 0 00.18-.109.236.236 0 00.063-.168v-.453a.262.262 0 00-.063-.176.416.416 0 00-.176-.113.977.977 0 00-.28-.039H8.214v-.625h1.402c.383 0 .707.031.98.098.274.066.481.164.622.297a.61.61 0 01.21.465v.648a.58.58 0 01-.214.457c-.145.125-.352.223-.621.289a4.207 4.207 0 01-.977.098H8.215zm0 0M13.816 6.324h.926l1.84 3.844h-1.305l-.996-2.52-.996 2.52h-1.308zm-.59 2.633h2.102v.613h-2.101zm0 0M19.273 8.695v1.473h-1.316V8.695l-1.441-2.37h1.367l.722 1.593.688-1.594h1.367zm0 0M23.156 6.324h1.32v3.844h-1.32zm.754 3.227h1.156c.16 0 .286-.02.38-.063.09-.043.132-.105.132-.183v-.516a.218.218 0 00-.062-.16.393.393 0 00-.184-.106 1.026 1.026 0 00-.289-.035h-1.137v-.605h1.106a.764.764 0 00.363-.074c.086-.051.129-.122.129-.211v-.414c0-.079-.043-.137-.129-.18a.83.83 0 00-.36-.066h-1.109v-.614h1.242c.348 0 .645.031.891.094.25.062.438.152.566.27a.563.563 0 01.196.43v.413a.431.431 0 01-.137.309c-.086.086-.215.16-.383.219a2.7 2.7 0 01-.605.113c.258.02.476.058.66.113.18.059.32.137.418.23a.44.44 0 01.144.329v.523a.575.575 0 01-.191.438c-.129.117-.312.21-.559.273-.246.063-.539.094-.882.094h-1.36V9.55zm0 0M29.836 8.695v1.473H28.52V8.695l-1.446-2.37h1.371l.72 1.593.69-1.594h1.368zm0 0M7.707 11.852h1.625v4.71H7.707zm.926 3.957h1.422c.199 0 .355-.028.468-.079.11-.054.165-.128.165-.222v-.633a.28.28 0 00-.075-.2.47.47 0 00-.222-.124 1.225 1.225 0 00-.36-.043H8.637v-.746h1.355c.192 0 .34-.032.45-.094.105-.059.16-.145.16-.258v-.508c0-.093-.055-.168-.16-.222a1.051 1.051 0 00-.442-.078H8.637v-.75h1.527c.426 0 .79.035 1.094.109.305.078.535.187.695.332a.708.708 0 01.238.527v.508c0 .14-.054.266-.164.375-.109.11-.265.2-.472.27-.207.07-.457.117-.746.144.316.02.586.067.812.137.223.07.395.164.512.281a.54.54 0 01.183.402v.645a.696.696 0 01-.238.531c-.152.149-.383.258-.687.336-.301.074-.664.117-1.086.117H8.637v-.757zm0 0M15.113 11.852h1.133l2.266 4.71h-1.606l-1.222-3.09-1.227 3.09h-1.605zm-.722 3.226h2.582v.754H14.39zm0 0M19.215 11.852h1.27l2.23 3.144-.18.125v-3.27h1.477v4.711h-1.285l-2.204-2.964.168-.13v3.095h-1.476zm0 0M25.172 11.852h1.617v4.71h-1.617zm1.285 2.535l1.844-2.535h1.824l-3.672 3.96v-1.425zm.59-.399l1.11-.48 2.269 3.055h-1.774zm0 0"/></svg>

<br>
<!-- TODO: insert your icon here -->
<YOUR SVG CODE>
</br>
```

#### If the icons are intended for use by Shopify, please provide the following info:
Who are you working with at Shopify? (avoid adding personal details, provide github handle(preferred) or first name and last name)

What's the expected date of this change to deploy on Shopify?

dannye0231 commented:
@diasnath can you please address the following issues in your SVG's?

PaymentIconTest#test_Payment_icon_SVGs_are_a_single_line [test/unit/payment_icon_test.rb:127]:
The 'cofidis3x' SVG file should contain a single line of markup, optionally terminated by an empty line

rails test /home/runner/work/payment_icons/payment_icons/test/unit/payment_icon_test.rb:124

..F

Failure:
PaymentIconTest#test_Every_payment_SVG_meets_accessibility_requirements [test/unit/payment_icon_test.rb:78]:
{:message=>"The 'cofidis3x' SVG file does not have the appropriate <title> value"}.
Expected: "Cofidis en 3 fois"
  Actual: "cofidis4x"

rails test /home/runner/work/payment_icons/payment_icons/test/unit/payment_icon_test.rb:61

claudiomunoz reviewed (changes requested):
Thank you for your submission. Please review the [Appearance guidelines ](https://github.com/activemerchant/payment_icons/blob/master/CONTRIBUTING.md#appearance) and resubmit

dannye0231 commented:
Hi @diasnath , any updates?  We will be running another merge in a few weeks time and will need your changes soon

dannye0231 commented:
There are also some errors showing up but I have no way to check it further as I do not have rights to your branch.  Are you able to provide me with access?

dannye0231 commented:
@diasnath can you check your SVG for the following issue?

Failure:
PaymentIconTest#test_Every_payment_SVG_meets_accessibility_requirements [test/unit/payment_icon_test.rb:78]:
{:message=>"The 'ipwireinst' SVG file does not have the appropriate <title> value"}.
Expected: "Ipwireinst"
  Actual: "Virement SEPA Instantané"

bin/rails test /home/runner/work/payment_icons/payment_icons/test/unit/payment_icon_test.rb:61

......

dannye0231 commented:
@diasnath there is another error in an SVG:

Failure:
PaymentIconTest#test_Every_payment_SVG_meets_accessibility_requirements [test/unit/payment_icon_test.rb:78]:
{:message=>"The 'cofidis3x' SVG file does not have the appropriate <title> value"}.
Expected: "Cofidis en 3 fois"
  Actual: "Cofidis3x"

dannye0231 commented:
@diasnath can you please check the following SVG for this issue?

Failure:
PaymentIconTest#test_Every_payment_SVG_meets_accessibility_requirements [test/unit/payment_icon_test.rb:78]:
{:message=>"The 'cofidis4x' SVG file does not have the appropriate <title> value"}.
Expected: "Cofidis en 4 fois"
  Actual: "Cofidis4x"

rails test /home/runner/work/payment_icons/payment_icons/test/unit/payment_icon_test.rb:61

dannye0231 commented:
Can you please check this error as well?


Failure:
PaymentIconTest#test_Every_payment_SVG_meets_accessibility_requirements [test/unit/payment_icon_test.rb:78]:
{:message=>"The 'ipwire' SVG file does not have the appropriate <title> value"}.
Expected: "Virement SEPA"
  Actual: "Ipwire"

rails test /home/runner/work/payment_icons/payment_icons/test/unit/payment_icon_test.rb:61

.....

dannye0231 commented:
There is another error:

Failure:
PaymentIconTest#test_Every_payment_SVG_meets_accessibility_requirements [test/unit/payment_icon_test.rb:78]:
{:message=>"The 'ipwireinst' SVG file does not have the appropriate <title> value"}.
Expected: "Virement SEPA Instantané"
  Actual: "Ipwireinst"

rails test /home/runner/work/payment_icons/payment_icons/test/unit/payment_icon_test.rb:61

diasnath commented:
Hello @dannye0231 
I don't really understand the rules for naming. In this error : https://github.com/activemerchant/payment_icons/pull/1298#issuecomment-2748598438 it said the opposite of the last one. What is the right text to put ?

dannye0231 commented:
Looks good now

diasnath commented:
Hello,
I don't understand the first point, there is a title associated to the id of aria-labelledby.
I also can't optimize more the SVG.
For the opacity i'm pushing the modification to be more readable.

dannye0231 commented:
Hi @diasnath , please follow our Contribution guidelines and the SVGs still need to be optimized.  You can use a tool like SVGO to perform this.

dannye0231 commented:
Can you please check the following error within cofidis3x:

The aria-labelledby attribute references "pi-cofidis3x," which is not defined in a <title> or <desc> element. This is a significant accessibility issue that needs to be addressed.
Status: ❌ (missing title/description for accessibility)

Failure:
PaymentIconTest#test_Every_payment_SVG_meets_accessibility_requirements [test/unit/payment_icon_test.rb:68]:
{:message=>"The 'cofidis3x' SVG file should have a single <title> tag"}.
Expected: 1
  Actual: 0

rails test /home/runner/work/payment_icons/payment_icons/test/unit/payment_icon_test.rb:61

.F

dannye0231 reviewed (approved):
SVG Review
1. Role Attribute:

✅ The role="img" attribute is correctly included, providing semantic meaning.
2. Dimensions:

✅ The width="38" and height="24" attributes are specified, ensuring the SVG has defined dimensions.
3. ViewBox:

✅ The viewBox="0 0 38 24" attribute is present, allowing for proper scaling and responsiveness.
4. Title Element:

✅ The <title> element is included with the ID id="pi-ipwireinst", providing a clear text alternative for accessibility.
5. Fill and Stroke:

✅ The SVG uses fill attributes appropriately. The background path has a white fill (fill="#fff"), and the second path has an opacity defined for the outline.
6. Color Classes:

✅ The use of CSS classes for gradient fills is appropriate and maintains consistency.
7. Path Definitions:

✅ The paths are defined correctly and utilize fill-rule="nonzero" as required.
8. Accessibility:

✅ The title provides context for screen readers, improving accessibility.
9. Structure:

✅ The SVG structure is well-organized, with proper use of groups and paths.
10. Color Contrast:

✅ The color contrast between the fill colors and the background appears sufficient for visibility and accessibility.

dannye0231 commented:
@diasnath can you please give me access to your branch?  I can't resolve a conflict which is coming up as I don't have access.

diasnath commented:
Is it good now ?

dannye0231 commented:
![logorequirements](https://github.com/user-attachments/assets/641172cb-354e-474f-aa96-35e564cb338c)

dannye0231 commented:
<img width="680" alt="image" src="https://github.com/user-attachments/assets/5ff87cf1-2b93-457b-9e00-bf05d1655f5f" />

dannye0231 commented:
Please refer to our Contribution guide as there are a few errors found in your latest submission:

❌ Inline Styles: The SVG uses a <style> tag, which should be avoided. Styles should be inline.
❌ Class Attributes: The SVG uses class attributes (st0, st3).

dannye0231 commented:
Can you please check the following issues which were flagged?

ID Attributes: ❌ There are no ID attributes in the provided SVG code.
Inline Styles: ❓ Styles are applied using both inline attributes (fill, style) and within a <style> tag, which is not recommended.

diasnath commented:
I fixed the inline style that were empty, but I didn't see style tags on my files. On which one exactly are they ?
Same about the ID attribute.

diasnath commented:
Hello, 
is there any news please ?

diasnath commented:
Hello @dannye0231 sorry for the delay I was on vacation.
Here is the modification needed, it should now be correct. I deleted the white background and changed the ids.

dannye0231 commented:
@diasnath please refer to our [Contributions](https://github.com/activemerchant/payment_icons/blob/master/CONTRIBUTING.md) guide regarding the use of style tags

❌ Inline Styles: The SVG uses style attributes (e.g., style="fill:#d0103b;", style="fill:#fff;") on path elements. The guideline states: "Avoid using <style> tags; styles should be applied inline within the SVG elements."  While these are technically inline, the guideline often implies using presentation attributes (like fill="" directly) rather than the style attribute.
